spanner/spansql/parser.go

```diff
@@ -1304,9 +1310,64 @@ func (p *parser) parseDMLStmt() (DMLStmt, *parseError) {
 		}, nil
 	}
 
+	if p.eat("UPDATE") {
+		tname, err := p.parseTableOrIndexOrColumnName()
+		if err != nil {
+			return nil, err
+		}
+		u := &Update{
+			Table: tname,
+		}
+		// TODO: parse alias.
+		if err := p.expect("SET"); err != nil {
+			return nil, err
+		}
+		for {
+			ui, err := p.parseUpdateItem()
+			if err != nil {
+				return nil, err
+			}
+			u.Items = append(u.Items, ui)
+			if p.eat(",") {
+				continue
+			}
+			break
+		}
+		if err := p.expect("WHERE"); err != nil {
+			return nil, err
+		}
+		where, err := p.parseBoolExpr()
+		if err != nil {
+			return nil, err
+		}
+		u.Where = where
+		return u, nil
+	}
+
 	return nil, p.errorf("unknown DML statement")
 }
 
+func (p *parser) parseUpdateItem() (UpdateItem, *parseError) {
+	col, err := p.parseTableOrIndexOrColumnName()
+	if err != nil {
+		return UpdateItem{}, err
+	}
+	ui := UpdateItem{
+		Column: col,
+	}
+	if err := p.expect("="); err != nil {
+		return UpdateItem{}, err
+	}
+	if p.eat("DEFAULT") {
+		return ui, nil
```

Comment:
I think you need a special value for `DEFAULT`, as this implementation will make it impossible to set something to `NULL`.

Reply:
The Spanner docs say that `DEFAULT` will set the column values to NULL (I guess because Spanner doesn't have non-NULL defaults?), but here ui.Value==nil [from `X = DEFAULT`] is distinct from ui.Value==NullLiteral(0) [from `X = NULL`], so both are possible.